### PR TITLE
Prepare for new block variations 3/3:  block variation loading

### PR DESCRIPTION
**Pull request by carstingaxion**
Preparation to bring some block variations from the [gatherpress-block-playground](https://github.com/carstingaxion/gatherpress-block-playground) into the plugin as part of #626.

While current blocks can rely on `register_block_type()` to get everything in place, block variations need more manual attention.

### Description of the Change

With this PR, I try to outline a possible way to load all variations the same way.

Because most of the variations need some extra php to work, I went with a singleton, that gets automatically ...

~~copied from `/src` to `/build` and lives in the folder of the variation. For this reason the autoloader doesn’t know about the classes and that’s why I required them manually~~ 

... loaded from `includes/core/classes/blocks`.

This design-decision was made in a private chat between @mauteri and me.

~~Closes #~~

### Changelog Entry

> Added - Generalized the loading of block-variations

### Credits

Props @carstingaxion, @mauteri 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/GatherPress/gatherpress/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by carstingaxion** [comment]
He @mauteri could you please take a look at this idea of how to load the variations?

I based this branch on the work of #888, that’s why there are so many commits, but if possible, only focus on the last three. And don’t forget about the PR description ;)

thanks.

File: includes/core/classes/traits/class-block-variation.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+defined( 'ABSPATH' ) || exit; // @codeCoverageIgnore
+
+use Error;
+
+/**
+ * Common class that handles registering and enqueuing of assets.
+ *
+ * @since 1.0.0
+ */
+trait Block_Variation {
```
Comment (mauteri) [inline includes/core/classes/traits/class-block-variation.php:21]: I think we can simplify this a bit and not include a new trait. I'd add code like this to `class-assets.php`, get the variations directory from the file system and loop through. There's a lot of helpers you can use in `class-assets.php` for getting dependencies and such from the asset. Let me know what you think.

**Review by mauteri — CHANGES_REQUESTED** [review]

File: includes/core/classes/traits/class-block-variation.php (hunk 1 above)
Comment (carstingaxion) [reply includes/core/classes/traits/class-block-variation.php:21]: This sounds good.
Now that you mention the class-assets.php, I’m wondering that I haven’t thought about this. But yes, seems totally plausible for all this asset-related stuff.
Thank you.

**Comment by carstingaxion** [comment]
@mauteri Do you have any clue why the „semver“ package is missing now?  (Which makes all the workflows fail)

**Comment by mauteri** [comment]
> @mauteri Do you have any clue why the „semver“ package is missing now?  (Which makes all the workflows fail)

I may have accidentally excluded it when I tried to resolve conflicts with the develop branch. Sorry about that.

**Comment by carstingaxion** [comment]
> > @mauteri Do you have any clue why the „semver“ package is missing now?  (Which makes all the workflows fail)
> 
> I may have accidentally excluded it when I tried to resolve conflicts with the develop branch. Sorry about that.

No problem. I'll take a look.

**Comment by carstingaxion** [comment]
Im starting now and hope to get this finished.

**Review by mauteri — APPROVED** [review]

**Comment by carstingaxion** [comment]
Woooow! That was super fast again, @mauteri 
Thank you!

**Pull request merged**